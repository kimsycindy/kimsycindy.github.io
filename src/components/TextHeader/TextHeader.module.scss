@import '../../styles/variables';

.TextHeader {
  text-rendering: optimizeLegibility;
  text-align: center;

  @media screen and (min-width: $breakpoint-md) {
    text-align: left;
  }
}

.header {
  display: inline-block;
  color: $black;
  font-family: $font-header;
  line-height: $lineHeight-heading;
  font-weight: 400;
  padding: 20px 8px;
  text-transform: uppercase;
}

.h1 {
  composes: header;
  font-size: $fontSize-h1;
  background-color: $primary;
}

.h2 {
  composes: header;
  font-size: $fontSize-h2;
  background-color: $accent;
}

.h3 {
  composes: header;
  font-size: $fontSize-h3;
  background-color: $tertiary;
}

.h4,
.h5,
.h6 {
  composes: header;
  font-size: $fontSize-h4;
  background-color: $tertiary;
}
