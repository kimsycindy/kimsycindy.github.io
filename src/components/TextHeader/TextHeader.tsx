import React from 'react'
import styles from './TextHeader.module.scss'

interface Props {
  priority: 1 | 2 | 3 | 4 | 5 | 6
  children: string
}

type Heading = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

const TextHeader: React.FC<Props> = ({ priority, children }) => {
  const HeaderTag = `h${priority}` as Heading
  return (
    <HeaderTag className={styles.TextHeader}>
      <span className={styles[`h${priority}`]}>{children}</span>
    </HeaderTag>
  )
}

export default TextHeader
