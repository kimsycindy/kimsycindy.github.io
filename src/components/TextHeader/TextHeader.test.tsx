import React from 'react'
import renderer from 'react-test-renderer'

import TextHeader from './TextHeader'

describe('TextHeader', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<TextHeader priority={1}>Hello world!</TextHeader>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
