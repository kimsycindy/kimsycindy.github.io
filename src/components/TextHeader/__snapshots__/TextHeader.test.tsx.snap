// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextHeader renders correctly 1`] = `
<h1
  className="TextHeader"
>
  <span
    className="h1"
  >
    Hello world!
  </span>
</h1>
`;
